core/src/solvers/pipeline_planner.cpp

```diff
@@ -53,160 +53,175 @@ namespace moveit {
+    const moveit::planning_pipeline_interfaces::SolutionSelectionFunction& solution_selection_function)
+  : node_(node)
+  , pipeline_id_planner_id_map_(pipeline_id_planner_id_map)
+  , stopping_criterion_callback_(stopping_criterion_callback)
+  , solution_selection_function_(solution_selection_function) {
+	// If the pipeline name - pipeline map is passed as constructor argument, use it. Otherwise, it will be created in
+	// the init(..) function
+	if (!planning_pipelines.empty()) {
+		planning_pipelines_ = planning_pipelines;
 	}
+	// Declare properties of the MotionPlanRequest
+	properties().declare<uint>("num_planning_attempts", 1u, "number of planning attempts");
+	properties().declare<moveit_msgs::msg::WorkspaceParameters>(
+	    "workspace_parameters", moveit_msgs::msg::WorkspaceParameters(), "allowed workspace of mobile base?");
+
+	properties().declare<double>("goal_joint_tolerance", 1e-4, "tolerance for reaching joint goals");
+	properties().declare<double>("goal_position_tolerance", 1e-4, "tolerance for reaching position goals");
+	properties().declare<double>("goal_orientation_tolerance", 1e-4, "tolerance for reaching orientation goals");
+	// Declare properties that configure the planning pipeline
+	properties().declare<bool>("display_motion_plans", false,
+	                           "publish generated solutions on topic " +
+	                               planning_pipeline::PlanningPipeline::DISPLAY_PATH_TOPIC);
+	properties().declare<bool>("publish_planning_requests", false,
+	                           "publish motion planning requests on topic " +
+	                               planning_pipeline::PlanningPipeline::MOTION_PLAN_REQUEST_TOPIC);
+	properties().declare<std::unordered_map<std::string, std::string>>(
+	    "pipeline_id_planner_id_map", std::unordered_map<std::string, std::string>(),
+	    "Set of pipelines and planners used for planning");
+}
 
-	PlannerCache::PlannerID id(pipeline_ns, spec.adapter_param);
-
-	std::weak_ptr<planning_pipeline::PlanningPipeline>& entry = cache.retrieve(spec.model, id);
-	planning_pipeline::PlanningPipelinePtr planner = entry.lock();
-	if (!planner) {
-		// create new entry
-		planner = std::make_shared<planning_pipeline::PlanningPipeline>(spec.model, node, pipeline_ns,
-		                                                                PLUGIN_PARAMETER_NAME, spec.adapter_param);
-		// store in cache
-		entry = planner;
+bool PipelinePlanner::setPlannerId(const std::string& pipeline_name, const std::string& planner_id) {
+	// Only set ID if pipeline exists. It is not possible to create new pipelines with this command.
+	if (pipeline_id_planner_id_map_.count(pipeline_name) > 0) {
+		pipeline_id_planner_id_map_[pipeline_name] = planner_id;
 	}
-	return planner;
+	RCLCPP_ERROR(node_->get_logger(),
+	             "PipelinePlanner does not have a pipeline called '%s'. Cannot set pipeline ID '%s'",
+	             pipeline_name.c_str(), planner_id.c_str());
+	return false;
```

Comment:
Is this function reads like its missing a `return true` in the if statement?

Reply:
Oh, yes!